src/frontend/js/contexts/CourseProductContext/index.tsx

```diff
@@ -22,11 +32,17 @@ export interface CourseProductProviderProps {
 export const CourseProductProvider = ({
   courseCode,
   productId,
+  enrollmentId,
   children,
 }: PropsWithChildren<CourseProductProviderProps>) => {
   const value = useMemo(
-    () => ({ productId, courseCode, key: `${courseCode}+${productId}` }),
-    [productId, courseCode],
+    () => ({
+      productId,
+      courseCode,
+      enrollmentId,
+      key: enrollmentId ? `${enrollmentId}+${productId}` : `${courseCode}+${productId}`,
```

Comment:
Let's use `_` as separator instead of `+` to stick to a convention (snake_case).